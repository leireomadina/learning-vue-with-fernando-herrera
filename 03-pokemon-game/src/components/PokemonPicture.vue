<template>
  <div class="pokemon-container">
    <img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="Pokemon">

    <!-- <img class="fade-in" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="Pokemon"> -->
  </div>
</template>

<script>
export default {

}
</script>

<style scoped>

/* Pokemon Picture */
.pokemon-container {
  height: 200px;
  position: relative;
}

img {
  height: 200px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}

.hidden-pokemon {
  filter: brightness(0);
}

</style>>
