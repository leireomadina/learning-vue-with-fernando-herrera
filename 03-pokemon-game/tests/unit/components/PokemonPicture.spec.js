import { shallowMount } from "@vue/test-utils"
import PokemonPicture from '@/components/PokemonPicture'

describe('PokemonPicture componente', () => {

	test('debe hacer match con el snapshot', () => {
		const wrapper = shallowMount( PokemonPicture, {
			props: {
				pokemonId: 1,
				showPokemon: false
			}
		})

		expect( wrapper.html() ).toMatchSnapshot()
	})

	test('debe mostrar la imagen oculta y el pokemon 100', () => {
		const wrapper = shallowMount ( PokemonPicture, {
			props: {
				pokemonId: 100,
				showPokemon: false
			}
		})

		const [ image1, image2] = wrapper.findAll('img')

		expect( image1.exists() ).toBeTruthy()
		expect( image2 ).toBe(undefined)

		expect( image1.classes('hidden-pokemon') ).toBe(true)

		expect(image1.attributes('src')).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')
	})

	test('debe mostrar el pokemon si showPokemon:true', () => {
		const wrapper = shallowMount ( PokemonPicture, {
			props: {
				pokemonId: 100,
				showPokemon: true
			}
		})

		const image2 = wrapper.find('img.fade-in')

		expect( image2.exists() ).toBeTruthy()
	})

})