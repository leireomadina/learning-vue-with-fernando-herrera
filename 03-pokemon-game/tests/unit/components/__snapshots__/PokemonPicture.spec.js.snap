// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture componente debe hacer match con el snapshot 1`] = `
<div class="pokemon-container"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="Pokemon">
  <!--v-if-->
</div>
`;
